File: lecture_5/src/hello_ros_cpp/src/practice3.cpp
#include "rclcpp/rclcpp.hpp"

class LoggingNode: public rclcpp::Node
{
public:
    LoggingNode(): Node("logging_node"), i_(1)
    {
        timer_ = this->create_wall_timer(
            std::chrono::seconds(1),
            std::bind(&LoggingNode::timer_callback, this));
    }
private:
    void timer_callback()
    {
        RCLCPP_INFO(this->get_logger(), "Counted to %d", i_);
        if (i_ % 3 == 0)
            RCLCPP_INFO(this->get_logger(), "is divisible by 3.");
        else if (i_ % 5 == 0)
            RCLCPP_DEBUG(this->get_logger(), "is divisible by 5.");
        else if (i_ % 7 == 0)
            RCLCPP_WARN(this->get_logger(), "is divisible by 7.");
        else if (i_ % 11 == 0)
            RCLCPP_ERROR(this->get_logger(), "is divisible by 11.");
        else if (i_ % 13 == 0)
            RCLCPP_FATAL(this->get_logger(), "is divisible by 13.");
        i_++;
    }
    rclcpp::TimerBase::SharedPtr timer_;
    int i_;
};

int main(int argc, char *argv[])
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<LoggingNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: lecture_5/Lecture 5. Source code/hello_ros_py/hello_ros_py/practice3.py
import rclpy
from rclpy.node import Node

class LoggingNode(Node):
    def __init__(self):
        super().__init__('logging_node')
        self.i = 1
        self.timer =self.create_timer(1.0, self.timer_callback) # 1 second

    def timer_callback(self):
        self.get_logger().info(f"Counted to {self.i}")
        if self.i % 3 == 0:
            self.get_logger().info('is divisible by 3')
        if self.i % 5 == 0:
            self.get_logger().debug('is divisible by 5')
        if self.i % 7 == 0:
            self.get_logger().warning('is divisible by 7')
        if self.i % 11 == 0:
            self.get_logger().error('is divisible by 11')
        if self.i % 13 == 0:
            self.get_logger().fatal('is divisible by 13')

        self.i += 1

def main(args=None):
    rclpy.init(args=args)
    node = LoggingNode()
    rclpy.spin(node)

    node.destroy_node()
    rclpy.shutdown()


if __name__ =='__main__':
    main()
    


